File: ruby/high-scores/high_scores.rb
class HighScores
  attr_reader :scores

  def initialize(scores)
    @scores = scores
  end

  def latest
    @scores.pop
  end

  def personal_best
    @scores.max
  end

  def personal_top_three
    reverse = @scores.sort.reverse
    reverse[0, 3]
  end
end

File: python/high-scores/high_scores.py
def latest(scores):
    return scores[-1]


def personal_best(scores):
    return max(scores)


def personal_top_three(scores):
    return sorted(scores, reverse=True)[:3]


# personal_top_three([10, 30, 90, 30, 100, 20, 10, 0, 30, 40, 40, 70, 70])
# personal_top_three([40, 20, 40, 30])
# personal_top_three([30, 70])
# personal_top_three([30])
